packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -231,8 +267,12 @@ export class MirrorNodeClient {
     public async getBlocks(blockNumber?: number | string[], timestamp?: string, limitOrderParams?: ILimitOrderParams, requestId?: string) {
         const queryParamObject = {};
         this.setQueryParam(queryParamObject, 'block.number', blockNumber);
-        this.setQueryParam(queryParamObject, 'timestamp', timestamp);
-        this.setLimitOrderParams(queryParamObject, limitOrderParams);
+        if (timestamp) {
```

Comment:
These checks should be handled by `setQueryParam()` already.
If not move it into there